lib/ami-existence-poller.js

```diff
@@ -0,0 +1,82 @@
+const assert = require('assert');
+const {runAWSRequest} = require('./aws-request');
+const Iterate = require('taskcluster-lib-iterate');
+
+const log = require('./log');
+
+const BATCH_SIZE = 100;
+
+class AmiExistencePoller {
+
+  constructor({ec2, runaws = runAWSRequest, amiPollDelay = 25, state}) {
+    assert(typeof ec2 === 'object');
+    assert(typeof state === 'object');
+
+    this.ec2 = ec2;
+    this.state = state;
+    this.runaws = runaws;
+    this.amis = [];
+
+    this.iterator = new Iterate({
+      maxIterationTime: 1000 * 60 * 15,
+      watchDog: 1000 * 60 * 15,
+      maxFailures: 10,
+      waitTime: amiPollDelay,
+      handler: async(watchdog) => {
+        try {
+          await this.poll();
+        } catch (err) {
+          console.dir(err.stack || err);
+          monitor.reportError(err, 'Error polling pricing data');
+        }
+      },
+    });
+  };
+
+  start() {
+    return this.iterator.start();
+  }
+
+  stop() {
+    return this.iterator.stop();
+  }
+
+  async loadAmis(amis) {
+    this.amis = amis ? amis : (await this.state.listAmiUsage()).map(o => o.id);
+  }
+
+  async poll() {
+    assert(Array.isArray(this.amis), 'this.amis is not an array');
+    let result = [];
+
+    // parallelize this with Promise.all?
```

Comment:
We could, but we're trying to minimize our API request rate, and since this is not involved in responsiveness of the API, let's leave them sequential.